huaweicloud/services/vpn/resource_huaweicloud_vpn_customer_gateway.go

```diff
@@ -166,13 +196,20 @@ func resourceCustomerGatewayCreate(ctx context.Context, d *schema.ResourceData,
 }
 
 func buildCreateCustomerGatewayBodyParams(d *schema.ResourceData) map[string]interface{} {
```

Comment:
You can change it into one function, buildCreateCustomerGatewayDeprecatedCustomerGatewayChildBody
and buildCreateCustomerGatewayCustomerGatewayChildBody is unnecessary.